crates/ruff_python_ast/src/source_code/comment_ranges.rs

```diff
@@ -25,6 +25,17 @@ impl CommentRanges {
             })
             .is_ok()
     }
+
+    /// Returns the comments who are within the range
+    pub fn comments_in_range(&self, range: TextRange) -> &[TextRange] {
+        let start = self
+            .raw
+            .partition_point(|comment| comment.start() < range.start());
+        let end = self
+            .raw
+            .partition_point(|comment| comment.end() < range.end());
```

Comment:
Nit: The end must be after the start. 
```suggestion
        let end = start + self
            .raw[start..]
            .partition_point(|comment| comment.end() < range.end());
```